### PR TITLE
feat(injections): add printf format strings

### PR DESCRIPTION
**Pull request by ObserverOfTime**
Replaces #3573. I forked the parser so it's no longer limited to double-quoted strings.

### CODE REVIEW

File: runtime/queries/c/injections.scm [hunk 1]
```diff
@@ -4,6 +4,28 @@
...
+ (#any-of? @_function "fprintf" "fscanf" "fprintf_s" "fscanf_s"
+                      "sprintf" "sscanf" "sscanf_s")
+ (#set! injection.language "printf"))
+
+((call_expression
+  function: (identifier) @_function
+  arguments: (argument_list
+               (_) . (_) . (string_literal (string_content) @injection.content)))
+ (#any-of? @_function "snprintf" "sprintf_s" "snprintf_s")
+ (#set! injection.language "printf"))
```
Comment (amaanq) [inline runtime/queries/c/injections.scm:27]: there are a lot more than these

`man 3 printf` 

Missing:
dprintf (2nd arg)
vprintf (1st)
vfprintf (2nd)
vdprintf (2nd)
vsprintf (2nd)
vsnprintf (3rd)

File: runtime/queries/printf/highlights.scm [hunk 2]
```diff
@@ -0,0 +1 @@
+(format) @character
```
Comment (amaanq) [inline runtime/queries/printf/highlights.scm:1]: `@character.special`?

Comment (ObserverOfTime) [reply runtime/queries/printf/highlights.scm:1]: I kinda want to differentiate format strings from other special characters.

Comment (amaanq) [reply runtime/queries/printf/highlights.scm:1]: I'd like that too, but I feel character and string are somewhat oftenly used as the same color in colorschemes

A custom capture wouldn't be bad imo, like `@character.format`

Comment (ObserverOfTime) [reply runtime/queries/printf/highlights.scm:1]: A custom capture would be pointless since `@character.printf` can be used anyway.

Comment (amaanq) [reply runtime/queries/printf/highlights.scm:1]: Right, this is fine then

File: runtime/queries/c/injections.scm (hunk 1 above)
Comment (ObserverOfTime) [reply runtime/queries/c/injections.scm:27]: Just gonna pretend MS CRT is not a thing.

**Comment by amaanq** [comment]
I was gonna jot more down but forgot, this has a pretty good list

https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/vsprintf-vsprintf-l-vswprintf-vswprintf-l-vswprintf-l?view=msvc-170

**Review by amaanq — APPROVED** [review]

**Comment by amaanq** [comment]
Added a few more missing ones, merge if you think it's good

**Comment by clason** [comment]
Was it intentional to only add this to the `main` branch?

**Pull request merged**